homeassistant/components/esphome/__init__.py

```diff
@@ -222,6 +222,7 @@ async def send_home_assistant_state_event(event: Event) -> None:
             # Only communicate changes to the state or attribute tracked
             if (
                 "old_state" in event.data
+                and event.data["old_state"] is not None
                 and "new_state" in event.data
```

Comment:
Should this be guarded as well?

Reply:
I haven't seen any reports of new_state being None, nor could I trigger that myself. That makes sense, as this method is called when there's a new state.